### PR TITLE
[chloggen] Use go embed to embed the template file

### LINKED ISSUE #224: chloggen panics on execution
chloggen panics when called to generate the changelog with this build:
https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953803247/jobs/6770466495

See this stacktrace:
```
panic: open go.opentelemetry.io/build-tools/chloggen@v0.3.0/internal/chlog/summary.tmpl: no such file or directory

goroutine 1 [running]:
text/template.Must(...)
	text/template/helper.go:26
go.opentelemetry.io/build-tools/chloggen/internal/chlog.summary.String({{0x5ebe97, 0x5}, {0xc000146400, 0x12, 0x20}, {0xc000197d70, 0x1, 0x1}, {0x0, 0x0, ...}, ...})
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/internal/chlog/summary.go:66 +0x311
go.opentelemetry.io/build-tools/chloggen/internal/chlog.GenerateSummary({0x5ebe97?, 0x51?}, {0xc00014[8](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:9)000, 0x2d, 0xc000026240?})
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/internal/chlog/summary.go:60 +0x73f
go.opentelemetry.io/build-tools/chloggen/cmd.update({{0xc000026004, 0x51}, {0xc000026180, 0x5e}, {0xc000026240, 0x5b}, {0xc000020540, 0x6[9](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:10)}}, {0x5ebe97, 0x5}, ...)
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/cmd/update.go:56 +0xc7
go.opentelemetry.io/build-tools/chloggen/cmd.glob..func2(0x758e20?, {0x5ebbc7?, 0x1?, 0x1?})
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/cmd/update.go:42 +0x70
github.com/spf13/cobra.(*Command).execute(0x758e20, {0xc000030880, 0x1, 0x1})
	github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x758b40)
	github.com/spf13/cobra@v1.6.1/command.go:[10](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:11)44 +0x3bd
github.com/spf[13](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:14)/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.6.1/command.go:968
go.opentelemetry.io/build-tools/chloggen/cmd.Execute()
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/cmd/root.go:35 +0x25
main.main()
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/main.go:22 +0x[17](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:18)
make: *** [Makefile:[24](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:25)9: chlog-preview] Error 2
Error: Process completed with exit code 2.
```

### PR DESCRIPTION
**Pull request by atoulme**
Fixes #224

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

File: chloggen/internal/chlog/summary.go [hunk 1]
```diff
@@ -16,12 +16,15 @@ package chlog
...
 	"bytes"
+	"embed"
 	"fmt"
-	"path/filepath"
 	"sort"
 	"text/template"
 )
 
+//go:embed summary.tmpl
+var tmpl embed.FS
```
Comment (bogdandrutu) [inline chloggen/internal/chlog/summary.go:26]:
```suggestion
//go:embed summary.tmpl
var tmpl []byte
```

File: chloggen/internal/chlog/summary.go [hunk 2]
```diff
@@ -61,13 +64,11 @@ func GenerateSummary(version string, entries []*Entry) (string, error) {
...
 
 func (s summary) String() (string, error) {
-	summaryTmpl := filepath.Join(moduleDir(), "summary.tmpl")
-
 	tmpl := template.Must(
 		template.
 			New("summary.tmpl").
 			Option("missingkey=error").
-			ParseFiles(summaryTmpl))
+			ParseFS(tmpl, "summary.tmpl"))
```
Comment (bogdandrutu) [inline chloggen/internal/chlog/summary.go:71]:
```suggestion
			Parse(string(tmpl))
```

**Comment by atoulme** [comment]
ah bleh. I'll fix.

**Comment by atoulme** [comment]
codecov says:
> [2023-01-19T01:18:03.580Z] ['verbose'] Passed token was 0 characters long

I think something is broken with the codecov secret.

**Review by bogdandrutu — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Pull request merged**